core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactoryImpl.java

```diff
@@ -658,23 +657,26 @@ public SqlTypeName getSqlTypeName() {
     private final StructKind kind;
     private final List<String> names;
     private final List<RelDataType> types;
+    private final boolean nullable;
```

Comment:
Before introducing this change the cache didn't distinguish between nullable and not nullable types. Was it a bug or there was another reason behind it?

Reply:
It was not a bug, it was just that this Key was used to create a RelRecordType (see line 73), and before this change the RelRecordType did not have a nullable instance variable, so there was no need to have a nullable in this Key

Reply:
OK, I see, thanks for the explanation!